### PR TITLE
ci: build wda for simulators in a task

### PR DESCRIPTION
**Pull request by KazuCocoa**
Only for https://github.com/appium/WebDriverAgent/actions/workflows/wda-package.yml for now from https://github.com/appium/WebDriverAgent/pull/689 to reduce scope. This tasks it not for a release task, so it is safe to merge separately.

The current release task depends on semantics release, so to tune this task in a release task such as parallel running, we need to learn GitHub Actions further about how to manage module updates. It may take more time, so I won't keep https://github.com/appium/WebDriverAgent/pull/689 as a PR only for it.

### CODE REVIEW

File: .github/workflows/wda-package.yml [hunk 1]
```diff
@@ -57,3 +71,106 @@ jobs:
...
+      ZIP_PKG_NAME_TVOS_SIM_X86_64: "WebDriverAgentRunner_tvOS-Build-Sim-x86-64.zip"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - uses: maxim-lobanov/setup-xcode@v1
+        with:
+          xcode-version: "${{ env.XCODE_VERSION }}"
+      - name: Build iOS for simulator for x86_64
+        run: sh $GITHUB_WORKSPACE/Scripts/ci/build.sh
```
Comment (mykola-mokhnach) [inline .github/workflows/wda-package.yml]: I think it makes sense to combine build and zip into a single script as you anyway need to run them both. Then put matrix parameters into the script as each build could be done in parallel

Comment (KazuCocoa) [reply .github/workflows/wda-package.yml]: yea, still tweaking them

**Comment by KazuCocoa** [comment]
https://github.com/appium/WebDriverAgent/actions/runs/5119421857

**Comment by KazuCocoa** [comment]
Ok, built packages were expected.

File: .github/workflows/wda-package.yml [hunk 2]
```diff
@@ -57,3 +60,55 @@ jobs:
...
           path: "${{ env.ZIP_PKG_NAME_TVOS }}"
+
+  for_simulator_devices_ios:
+    needs: [host_machine]
+    name: Build WDA for iOS simulators
+    runs-on: ${{ needs.host_machine.outputs.host }}
+
+    strategy:
+      matrix:
+        arch: [x86_64, arm64]
```
Comment (mykola-mokhnach) [inline .github/workflows/wda-package.yml:77]: can we also move tvos/ios into the matrix?

File: .github/workflows/wda-package.yml [hunk 3]
```diff
@@ -78,10 +78,15 @@ jobs:
         with:
           xcode-version: "${{ env.XCODE_VERSION }}"
       - name: Create a zip of WebDriverAgentRunner${{ matrix.target }} for simulator for ${{ matrix.arch }}
-        run: sh $GITHUB_WORKSPACE/Scripts/ci/build-sim.sh
+        run: |
+          if [[ $TARGET == '_tvOS' ]]; then
```
Comment (mykola-mokhnach) [inline .github/workflows/wda-package.yml]: I assume we could directly use ` ${{ matrix.target }} ` in bash scripts

File: .github/workflows/wda-package.yml [hunk 4]
```diff
@@ -78,10 +78,15 @@ jobs:
...
+            DESTINATION=$DESTINATION_SIM_TVOS sh $GITHUB_WORKSPACE/Scripts/ci/build-sim.sh
```
Comment (mykola-mokhnach) [inline .github/workflows/wda-package.yml]: so if let say `matrix.target == iOS` we could build the variable name dynamically.
See https://github.com/orgs/community/discussions/25768 and https://stackoverflow.com/questions/16553089/dynamic-variable-names-in-bash

**Comment by KazuCocoa** [comment]
maybe its better for the matrix

File: .github/workflows/wda-package.yml [hunk 5]
```diff
@@ -2,53 +2,60 @@ name: Building WebDriverAgent
 
 on:
   workflow_dispatch:
+  workflow_run:
+    workflows: ["Release"]
+    types:
+      - completed
```
Comment (KazuCocoa) [inline .github/workflows/wda-package.yml:8]: This kicks https://github.com/appium/WebDriverAgent/actions/workflows/wda-package.yml after each release. Still the upload destination is not GitHub releases https://github.com/appium/WebDriverAgent/releases, though. (Eventually the upload destination want to be the releases page)

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**